tests/eval_files/async912.py

```diff
@@ -0,0 +1,108 @@
+# ASYNCIO_NO_ERROR
+import trio
+
+
+async def foo():
+    with trio.move_on_after(0.1):  # error: 4
+        ...
+    with trio.move_on_at(0.1):  # error: 4
+        ...
+    with trio.fail_after(0.1):  # error: 4
+        ...
+    with trio.fail_at(0.1):  # error: 4
+        ...
+    with trio.CancelScope(0.1):  # error: 4
+        ...
+
+    with open(""):
+        ...
+
+    with trio.move_on_after(0.1):
+        await trio.lowlevel.checkpoint()
+
+    with trio.move_on_after(0.1):  # error: 4
+        with trio.move_on_after(0.1):  # error: 8
+            ...
+
+    with trio.move_on_after(0.1):  # TODO: should probably raise an error?
+        with trio.move_on_after(0.1):
```

Comment:
Because the inner `move_on_*` swallows the cancellation?

I think this would be prone to false alarms, because it depends on which one triggers first (if the outer, we're fine, for example).  I'm concerned that trying to be fancy would disallow tricky-but-valid cases, so let's just leave this one alone and accept some missed alarms.